consensus/src/consensus/mod.rs

```diff
@@ -842,10 +870,73 @@ impl ConsensusApi for Consensus {
         })
     }
 
+    fn get_block_even_if_header_only(&self, hash: Hash) -> ConsensusResult<Block> {
+        let Some(status) = self.statuses_store.read().get(hash).unwrap_option().filter(|&status| !status.has_block_header()) else {
```

Comment:
pretty sure you want `.filter(|&status| status.has_block_header())` without `!` negation

Reply:
pretty sure I agree :)
nice catch!